src/app/modules/weight/components/weight-progress/weight-progress.component.ts

```diff
@@ -0,0 +1,46 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnDestroy } from '@angular/core';
+import { combineLatest, Subject } from 'rxjs';
+import { filter, shareReplay, takeUntil } from 'rxjs/operators';
+import { WeightProgressRemaining } from '../../models/weight';
+import { WeightService } from '../../weight.service';
+
+@Component({
+  selector: 'app-weight-progress',
+  templateUrl: './weight-progress.component.html',
+  styleUrls: ['./weight-progress.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class WeightProgressComponent implements OnDestroy {
+  weight$ = this.weightService.getWeight().pipe(shareReplay(1));
+  remaining: WeightProgressRemaining;
+  private unsubscribe$: Subject<void> = new Subject();
+  constructor(
+    private readonly weightService: WeightService,
+    private cdr: ChangeDetectorRef
+  ) {
+    combineLatest([
+      this.weight$,
+      this.weightService.getGoal()
+    ])
+      .pipe(
+        filter(([weight, goal]) => weight.length && goal >= 0),
+        takeUntil(this.unsubscribe$)
+      ).subscribe(([weight, goal]) => {
+        const latestWeight = weight[weight.length - 1].weight;
+        const value = latestWeight - goal;
+        const percent = goal / latestWeight * 100;
+
+        this.remaining = { value, percent };
+        // Marking for check because `remaining` variable is used in view
+        // and we need to render this value. There are no other triggers that
+        // trigger change detection and we set this value only when there is goal and
+        // we have weight records
+        this.cdr.markForCheck();
```

Comment:
Right. Do you do this manual optimization a lot?

Reply:
You mean manual optimization with `OnPush` strategy or with `markForCheck` / `detectChanges` in general ? If `OnPush` then I try to use it as much as I can to be in control of change detection and provide better performance because its not really that hard. I don't know how much of a difference it would be in a large project. If its more about `markForCheck` then I use it only when I see that something is not triggering the detection and thus I need to manually do it. Hope I've answered your question :)